### PR TITLE
MedicationProfiles

### PR DESCRIPTION
**Pull request by ArdonToonstra**
Hi @pieter-edelman-nictiz , 
A new PR for MP. Compared to last time, these are all the changes I have made: https://github.com/Nictiz/Nictiz-R4-zib2020/blob/MedicationProfiles/mp%20changes%20since%20validation%20round3.md

I think the profiles need some special attention to the status descriptions and StopType mapping. It's a biggie this PR... Sorry about that.

Edit: Forge does not open the extension on a polymorphic element while it is in the XML. I made a bug report.

### CODE REVIEW

**Comment by ArdonToonstra** [comment]
@pieter-edelman-nictiz , @jduwel  please help me out. I can't get this any further while I would like to have these profile in main for the upcoming release.

#### ❌ "Validate zib profiles" failed with 0 fatal messages, 1 errors, 2 warnings and 1 information messages.
Error is about terminology, not able to find the right code. Warnings about missing profile references.

#### ❌ "Validate nl-core profiles" failed with 0 fatal messages, 1 errors, 14 warnings and 7 information messages.
Error: the purpose field does not match ProfilingGuidelines. This is because this is a profile without a zib. The QA tooling should be adjusted. Warnings because of missing profiles.

#### ❌ "Validate other profiles" failed with 0 fatal messages, 3 errors, 1 warnings and 1 information messages.
Error: SearchParameter are not yet supported by QA tooling.
Warning: missing profile.

#### ❌ "Validate other terminology" failed with 0 fatal messages, 4 errors, 29 warnings and 0 information messages.
Error: NullFlavor system exported in the ValueSets is not correct. Asked Alexander Henket.

#### ❌ "Validate examples" failed with 0 fatal messages, 8 errors, 33 warnings and 26 information messages.
Errors: all terminology related. 
Warnings: all terminology related.

Zib compliance currently does not work. Moreover, I wasn't able to silence the known-issues for InstructionsForUse.DosageInstruction while I did include them in the known-issues.yml file.

**Comment by pieter-edelman-nictiz** [comment]
> Error: the purpose field does not match ProfilingGuidelines. This is because this is a profile without a zib. The QA tooling should be adjusted. Warnings because of missing profiles.

This is a situation that is not accounted for. It's not trivial to adjust the tooling without laying some guidelines first.
As far is I'm concerned, it's perfectly valid to prefix any profile at the nl-core layer with "nl-core", without regard for the fact if there's a zib.

**Comment by pieter-edelman-nictiz** [comment]
> Error: SearchParameter are not yet supported by QA tooling.
Warning: missing profile.

I'd suggest to postpone this until we've found some new naming conventions

**Comment by pieter-edelman-nictiz** [comment]
> ❌ "Validate zib profiles" failed with 0 fatal messages, 1 errors, 2 warnings and 1 information messages.
> 
> Error is about terminology, not able to find the right code. Warnings about missing profile references.

This is not blocking for the PR.

**Comment by pieter-edelman-nictiz** [comment]
> ❌ "Validate examples" failed with 0 fatal messages, 8 errors, 33 warnings and 26 information messages.
> Errors: all terminology related.
> Warnings: all terminology related.

If you look at the required check, you'll see that there actually is a code that's not from the required ValueSet. Should be fixed in ADA.

File: resources/zib/zib-MedicationAdministration2.xml [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="url" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+      <min value="0" />
+    </element>
+    <element id="MedicationAdministration.dosage.rate[x].extension:range">
```
Comment (pieter-edelman-nictiz) [inline resources/zib/zib-MedicationAdministration2.xml:342]: @ArdonToonstra the zib compliance tool trips on this piece of code. Shouldn't the extension be in the `:rateRange` slice?

**Comment by ArdonToonstra** [comment]
> > ❌ "Validate examples" failed with 0 fatal messages, 8 errors, 33 warnings and 26 information messages.
> > Errors: all terminology related.
> > Warnings: all terminology related.
> 
> If you look at the required check, you'll see that there actually is a code that's not from the required ValueSet. Should be fixed in ADA.

Missed that one. Fixed.

File: resources/zib/zib-MedicationAdministration2.xml (hunk 1 above)
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationAdministration2.xml]: Didn't get it to work. Removed it all. Forge can't really handle it as well. I think I was bending FHIR to it's limits.

**Comment by ArdonToonstra** [comment]
Still needs to silence the compliancy for InstructionsForUse.DosageInstruction but I can't see what I am doing wrong.

**Comment by jduwel** [comment]
Remarks regarding latest QA checks:

- Everything involving `differential.element[0]` ignored. This is due to a validator change
- zib-Encounter: reference to targetProfile zib-LaboratoryTestResult cannot be resolved. MP9 refers to nl-core-Encounter by business identifier, but the Encounters themselves will hardly be resolved (nor does MP9 expects them to be), so I am not expecting practical issues.
- zib-NursingIntervention refering to zib-TreatmentObjective: not used within MP9
- zib-MedicalDevice.Product, zib-VisualFunction.VisualAid, zib-HearingFunction.HearingAid, zib-AllergyIntolerance: correct errors on sd-pg-03 (some seem genuine, some involve reversed references that perhaps should be added as an exception to QA). However, these do not impact MP9, so I am inclined to not solve these errors here
- zib-ContactInformation-EmailAddresses, zib-Patient, zib-AddressInformation, zib-HealthcareProvider: correct errors on sd-pg-03. TBD what to do with these.
- zib-HearingFunction.HearingAid, zib-HearingFunction.HearingAid.Product, zib-VisualFunction.VisualAid.Product -  Terminology warning on ValueSet expansion: seems incorrect, the ValueSet referenced are available.

**Pull request merged**